contracts/FROST.sol

```diff
@@ -266,7 +281,7 @@ library FROST {
         // to a curve scalar, there are some values of `z` that can be
         // specified in more than one way.
         {
-            bool pOk = _isOnCurve(px, py);
+            bool pOk = isValidPublicKey(px, py);
             bool rOk = _isOnCurve(rx, ry);
```

Comment:
Isn't this the case because `z` is, by computation, an element of F[n] and not because this is required by `ecrecover`?